optimum/habana/checkpoint_utils.py

```diff
@@ -61,15 +62,14 @@ def get_checkpoint_files(model_name_or_path, local_rank, token=None):
     return file_list
 
 
-def write_checkpoints_json(model_name_or_path, local_rank, checkpoints_json, token=None):
+def write_checkpoints_json(model_name_or_path, local_rank, f, token=None):
     """
     Dumps metadata into a JSON file for DeepSpeed-inference.
     """
-    checkpoint_files = get_checkpoint_files(model_name_or_path, local_rank, token=token)
-    if local_rank == 0:
-        data = {"type": "ds_model", "checkpoints": checkpoint_files, "version": 1.0}
-        with open(checkpoints_json, "w") as fp:
-            json.dump(data, fp)
+    checkpoint_files = get_checkpoint_files(model_name_or_path, local_rank, token)
+    data = {"type": "ds_model", "checkpoints": checkpoint_files, "version": 1.0}
+    json.dump(data, f)
+    f.flush()
```

Comment:
With DeepSpeed and several devices, all processes will write tot this json file no?

Reply:
no, because i'm using temp file, so every process create different file and dump it.

Reply:
@yeonsily Would that change work with the way you use this method in TGI?

Reply:
i don't see anyone using this code right now.
if you are referring to this PR - https://github.com/huggingface/optimum-habana/pull/485/files
so yes it should get updated same as my change..